geps/gep-1619.md

```diff
@@ -163,43 +163,258 @@ Java application servers such as Tomcat and Jetty, were the first to standardize
 
 ### Session Persistence API
 
-The majority of implementations and dataplanes support configuring cookies for session persistence while support for header-based session persistence is limited.
+Although some implementations and dataplanes support session persistence through headers, cookies are more universally
+supported and portable; therefore, this GEP will focus on implementing cookie-based session persistence.
 
-To organize potential Gateway API configuration of cookie functionality, the next section will break down configuration into specific categories:
-- Session Cookie Configuration
-- Session Cookie Strategy
+To determine how the session persistence API should be designed, we must consider the following:
+1. Who is responsible for configuring it?
+2. What existing API object it is augmenting?
+3. What are the practical implications of attaching session persistence to an existing API object?
+4. How are existing implementation APIs designed?
 
-**Session Cookie Configuration**
+**Who is responsible for configuring it?**
 
-Session cookie configuration are the parameters and attributes the dataplane uses to create a cookie for establishing a session.
+Referring to the [Gateway API Security Model](https://gateway-api.sigs.k8s.io/concepts/security-model/#roles-and-personas),
+the target kubernetes role/persona for session persistence, are application developers. In most cases, session
+persistence is not considered an optional feature, but rather a necessary and deliberate component incorporated into the
+application's architecture by the developers themselves. Consequently, application developers should be responsible for
+adjusting the configuration to ensure the functionality of their applications.
 
-A set-cookie request can be broken down into [3 main components](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie):
-- Cookie Name
-- Cookie Value
-- Cookie Attributes
+**What existing API object it is augmenting?**
 
-Only the Cookie Name and Cookie Value are required. All Cookie Attributes are optional. Attributes are simply key=value pairs with the value as optional for some attributes.
+Session persistence augments a Kubernetes service object after the routing to the service has taken place. In
```

Comment:
That might be a common use-case but I would like to point out additional use-cases such as traffic splitting, canary deployment (https://istio.io/latest/docs/tasks/traffic-management/traffic-shifting/) where session persistence should still work. So the cookie should force a request to go to a particular Kubernetes Service disregarding the traffic splitting/shifting rules just like we expect the request to go to a particular backend disregarding load balancing within that Service. This is one of the use-cases we are interested in.

Reply:
To address my point, we should say that Session persistence may also augment an HTTPRoute (or GRPCRoute) where we indicate service stickyness based on the session cookie.

Reply:
That's a great point, I missed this use case. I agree it makes sense to have `HTTPRoute` and `GRCPRoute` be targets given this use case.

Now, I question whether there is still a good reason for `Services` to be a valid target. Seems like attaching to a `HTTPRoute` or `Service` should provide the same session persistence functionality, right? i.e. attaching to a `Service` should still have the `HTTPRoute` stick a session to the same service regardless of traffic shifting configuration?

Seems the only benefit here for `Service`, is, the practical matter that you can easily "reuse" the configuration among multiple `HTTPRoutes`.

If you can attach to an `xRoute`, I feel like it might confuse users why they should attach to a `Service` over a `xRoute` or visa-verse.

Reply:
I think there are other comments indicating 'reuse' is not the main reason for setting sessions on Service. 

If we want to avoid confusing users - we should probably only attach to service and defer per route. But I don't think it's a huge confusion - some servers will support both, others will only support service.

Reply:
Just to be clear, are we saying attaching to a Service vs. attaching to a HTTPRoute provides different functionality? i.e. would attaching to a Service still have the HTTPRoute stick a session to the same service regardless of traffic shifting configuration? Or would you have to attach to a HTTPRoute for persistence over traffic splitting?